plugins/inputs/diskio/diskio.go

```diff
@@ -125,9 +129,29 @@ func (d *DiskIO) Gather(acc telegraf.Accumulator) error {
 			"merged_reads":     io.MergedReadCount,
 			"merged_writes":    io.MergedWriteCount,
 		}
-		acc.AddCounter("diskio", fields, tags)
+		if d.lastIOCounterStat != nil {
+			if lastValue, exists := d.lastIOCounterStat[k]; exists {
+				deltaRWCount := float64(io.ReadCount + io.WriteCount - lastValue.ReadCount - lastValue.WriteCount)
+				deltaRWTime := float64(io.ReadTime + io.WriteTime - lastValue.ReadTime - lastValue.WriteTime)
+				deltaIOTime := float64(io.IoTime - lastValue.IoTime)
+				fieldsG := make(map[string]interface{})
+				if deltaRWCount > 0 {
+					fieldsG["io_await"] = deltaRWTime / deltaRWCount
+					fieldsG["io_svctm"] = deltaIOTime / deltaRWCount
+				}
+				itv := float64(collectTime.Sub(d.lastCollectTime).Milliseconds())
+				if itv > 0 {
+					fieldsG["io_util"] = 100 * deltaIOTime / itv
+				}
+				if len(fieldsG) > 0 {
+					acc.AddGauge("diskio", fieldsG, tags)
+				}
```

Comment:
This is because the other metrics are of type "Counter," while the newly added values are of type "Gauge"

Reply:
You are right, it is also acceptable if we don't strictly differentiate.

Reply:
Please don't remove fields from existing metrics as this will break everyones queries. An alternative would be to make the metric "untyped" if one of the new fields is included... What do you think?

Reply:
Good idea, then I will stick with the previous logic and just add new fields.